### ISSUE TITLE
dbc.TextArea n_submit should not increment for Shift+Enter?

### ISSUE THREAD

**Issue by TimChild** [facultyai/dash-bootstrap-components#967]
First, I'd like to say thanks for all the great work you have done here! I'm not 100% this is a bug, or a feature request, but it's hopefully pretty simple either way.

- dash version: `2.7.0`
- dash-bootstrap-components version: `1.4.2`
- components affected by bug: dbc.TextArea

### What is happening?

Both `Enter` and `Shift+Enter` increment `n_submit` for the dbc.TextArea.

### What should be happening?

I would expect that only `Enter` increments `n_submit` and `Shift+Enter` is ignored.

### Code
Lines 73-74 of dash-bootstrap-components/src/components/input/Textarea.js are currently:
```js
  const onKeyPress = e => {
    if (setProps && e.key === 'Enter') {
```

I expect the only chance necessary is to add a check that the buttons are not `Shift+Enter`, but I have absolutely zero experience with JS, so I figured it would be less work for everyone if I just made a bug report rather than a botched pull request.

Please let me know if I am wrong about what should be the default behavior here. 

Thanks again!

**Comment by tcbegley** [comment]
Hi @TimChild 

Thanks for raising this, and sorry for taking some time getting back to you. 

This is an interesting one, it seems the default behaviour for HTML forms is that `<textarea>` will not submit, whether or not shift is pressed, enter will always give a new line. By contrast `<input>` will submit the form if enter is pressed.

I think your suggestion makes the most sense, shift+enter makes new line, enter by itself submits. My only hesitation is that we would end up diverging slightly from the behaviour of dash-core-components.

I've implemented this change and made it available as a pre-release, I'd be interested to hear how you get on if you have a moment to try it with your app

```
pip install dash-bootstrap-components==1.5.0rc1
```

Here's a simple demo app you can try with if you don't have something to hand

```python
import dash_bootstrap_components as dbc
from dash import Dash, html, Input, Output

app = Dash(external_stylesheets=[dbc.themes.BOOTSTRAP])

app.layout = dbc.Container(
    [dbc.Textarea(id="textarea", debounce=True), html.P(id="output")],
    class_name="p-5",
)


@app.callback(Output("output", "children"), Input("textarea", "value"))
def return_value(value):
    return value


if __name__ == "__main__":
    app.run_server(debug=True)
```

**Comment by TimChild** [comment]
@tcbegley, No worries for the delay. Thanks for your considered response! 

Yes, I'd noticed that behaviour of `input` vs `textarea` in the `dcc` components, and I understand the hesitation to deviate from the `dcc.Textarea` behaviour too. At least for my use case, it is just so much more convenient to have the `Enter` to submit and `Shift + Enter` for newline (I've tested the pre-release and it works great). For me, that is the behaviour that I expected in the first place from the `dcc.Textarea` (I first assumed that it was just a multi-line `Input`), and I was disappointed when I found it did not have that. 

On the other hand, I can imagine scenarios when you would want `Enter` to just be treated as a newline, so there is also an argument for removing the submit on enter from your `dbc.Textarea` to allow for that and to re-align with the `dcc.Textarea` behaviour. 

My vote would be to go ahead with the `Enter` for submit and `Shift + Enter` for newline. It's intuitive, and very helpful when you do want to use that behaviour. 
In the case that someone really doesn't want the additional behaviour, could they just apply bootstrap styling to the `dcc.Textarea`, similar to your recommendation for the `dcc.Dropdown`? (https://github.com/tcbegley/dash-bootstrap-css)
